### PR TITLE
Feature/few shot

### PR DESCRIPTION
**Pull request by tvashtar**
Hi Jinshuai, 
I've merged main into this branch and resolved all conflicts on the branch, and tested the few shot learning notebook. 
This adds a couple of functions to enable few_shot learning. The primary ones are:
modules/prompts/get_few_shot_prompt
modules/analyze/bulk_analyze_text_few_shot

We could probably refactor these into the zero shot functions but I didn't for two reasons:
1. We had two work branches, and I was worried that would cause more merge conflicts
2. It would make these long functions even longer. 

The pull request diff, seems a little confused on the analyze functions, so we should test post merge. 

After the merge this I'll start a new branch for some work on the "No manifesto" queries that I have stashed.

### CODE REVIEW

**Comment by tvashtar** [comment]
I made one extra simpler change in analyze.py where I am now coding errors created by the code/invocation as 'ERR' leaving 'NA' only as a true response from the 'LLM'. I think most of the 'NA's we are getting were actually just the LLMs not returning a single value. However to fully show this we will need to rerun results.

File: modules/prompts.py [hunk 1]
```diff
@@ -58,3 +61,48 @@ def get_prompts(issue_area, text, override_persona_and_encouragement=None):
...
+    # were sufficient.
+
+    persona = personas[0]
+    encouragement = encouragements[1]
+
+    prompt = [SystemMessage(content=system_template.format(
+        persona=persona, encouragement=encouragement, policy_scale=policy_scales[issue_area]))]
+    for example in issue_examples.iterrows():
+        summary_file_name = example[1]['Calibration File']
+        with open(f'../data/summaries/{summary_file_name}', 'r') as summary_file:
```
Comment (Mayazure) [inline modules/prompts.py:99]: The summary files are not included in this repo.  Should we add those files or generate them first when setup the project locally?

**Review by Mayazure — CHANGES_REQUESTED** [review]
The previous codes work well on my local environment so the merge was a success.
The new few_shot codes require summary files which are not in data/summaries folder yet. Please see comment.

**Comment by tvashtar** [comment]
Yeah that’s a great question. I was intentionally not tracking them because
they change so frequently but with the few shot learning it does need them.
I can add them for now.

In the upcoming work I’ll also add checks if a summary exists and by
default doesn't recreate them.

In the long run once you know where this code and work will live it would
be good to create a proper database to store the results, run metadata and
either the summaries themselves or at least file paths to some central
storage (AWS S3).


Conor

**Pull request merged**